edb/tools/test/runner.py

```diff
@@ -124,20 +126,18 @@ def _run(self, test, result):
         return result
 
 
-def _run_test(case):
-    case_class = type(case)
-
-    try:
-        suite = suite_cache.cache[case_class]
-    except KeyError:
-        suite = suite_cache.cache[case_class] = StreamingTestSuite()
-
-    if isinstance(case, collections.abc.Iterable):
-        # Got a test suite
-        for test in case:
-            suite.run(test, result)
+def _run_test(workload):
+    if workload is teardown_suite:
```

Comment:
`is`? Maybe using `==` is safer in this case?

Reply:
Ah, this is actually unused. Removed.